Mathlib/Data/List/Basic.lean

```diff
@@ -1010,6 +1010,43 @@ theorem zipWith_flip (f : α → β → γ) : ∀ as bs, zipWith (flip f) bs as
 
 /-! ### take, drop -/
 
+@[simp] lemma take_eq_self (x : List α) n : x.take n = x ↔ x.length ≤ n :=
+  ⟨fun h ↦ by rw [← h]; simp; omega, take_of_length_le⟩
+
+@[simp] lemma take_self_eq (x : List α) n : x = x.take n ↔ x.length ≤ n := by
+  rw [Eq.comm, take_eq_self]
+
+@[simp] lemma take_eq_left {x y : List α} n :
+  (x ++ y).take n = x.take n ↔ y = [] ∨ n ≤ x.length := by
+  constructor
+  · rcases le_or_gt n x.length with _ | h
+    · tauto
+    · cases y
+      · tauto
+      · intro h; replace h := congr_arg length h; simp at h; omega
+  · rintro (rfl | h)
+    · rw [append_nil]
+    · exact take_append_of_le_length h
+
+@[simp] lemma take_eq_right {x y : List α} n :
+  x.take n = (x ++ y).take n ↔ y = [] ∨ n ≤ x.length := by
+  rw [Eq.comm]; apply take_eq_left
+
+@[simp] lemma drop_take_append_drop (x : List α) (m n : ℕ) :
+  (x.drop m).take n ++ x.drop (n + m) = x.drop m := by rw [← drop_drop, take_append_drop]
+
+/--order of summands is swapped-/
```

Comment:
I'd write this as
```suggestion
/-- Compared to `drop_take_append_drop`, the order of summands is swapped. -/
```
And I wonder: are really *both* lemmas useful, and should they both be simp lemmas?